includes/writer.js

```diff
@@ -47,6 +52,19 @@ module.exports = function(couchDbUrl, bufferSize, parallelism) {
       }
       cb();
     });
+
+    response.on('response', function(resp) {
+      if (resp.statusCode === 401) {
+        var e = new error.BackupError('Unauthorized', `Database ${couchDbUrl} does not have correct permissions. Check that you have the correct permissions before restoring.`);
+        response.abort();
+        writer.end();
+        // only emit the first error as there are multiple threads
+        if (!didError) {
```

Comment:
I don't think you need this. Let all the writers emit `error`. We appear only to handle the first and terminate.